core/benches/scheduler.rs

```diff
@@ -0,0 +1,204 @@
+
+    fn test_pre_lock_filter(_tx: &TransactionState<Tx>) -> PreLockFilterAction {
+        PreLockFilterAction::AttemptToSchedule
+    }
+}
+
+// TODO - support both schedulers
+fn bench_scheduler_impl<T: ReceiveAndBuffer + utils::ReceiveAndBufferCreator>(
+    c: &mut Criterion,
+    bench_name: &str,
+) where
+    <T as ReceiveAndBuffer>::Transaction: 'static,
+{
+    let mut group = c.benchmark_group("bench_greedy_scheduler");
+    group.sample_size(10);
+
+    let tx_counts: Vec<(usize, &str)> = vec![(16 * 1024, "16K_txs")]; //TOTAL_BUFFERED_PACKETS took too
+                                                                      //long
+    let ix_counts: Vec<(usize, &str)> = vec![
+        (1, "single_ix"),
+        (utils::MAX_INSTRUCTIONS_PER_TRANSACTION, "max_ixs"),
+    ];
+    // TODO - add tx accounts conflict config
+
+    for (ix_count, ix_count_desc) in &ix_counts {
+        for (tx_count, tx_count_desc) in &tx_counts {
+            let bench_name = format!("{bench_name}/{ix_count_desc}/{tx_count_desc}");
+            group.throughput(Throughput::Elements(*tx_count as u64));
+            group.bench_function(&bench_name, |bencher| {
+                bencher.iter_custom(|iters| {
+                    let utils::ReceiveAndBufferSetup {
+                        txs,
+                        sender,
+                        mut container,
+                        mut receive_and_buffer,
+                        decision,
+                    }: utils::ReceiveAndBufferSetup<T> =
+                        utils::setup_receive_and_buffer(*tx_count, *ix_count, 0.0, true);
+
+                    let mut execute_time: Duration = std::time::Duration::ZERO;
+                    for _i in 0..iters {
+                        // setup new Scheduler and Container for each iteration of execution
+                        let bench_env: BenchEnv<T::Transaction> = BenchEnv::new();
+                        let scheduler = GreedyScheduler::new(
+                            bench_env.consume_work_senders.clone(),
+                            bench_env.finished_consume_work_receiver.clone(),
+                            GreedySchedulerConfig::default(),
+                        );
+                        let mut scheduler = black_box(scheduler);
+
+                        // reset/refill container
+                        if sender.send(txs.clone()).is_err() {
+                            panic!("Unexpectedly dropped receiver!");
+                        }
+                        container.clear();
+                        let mut count_metrics = SchedulerCountMetrics::default();
+                        let mut timing_metrics = SchedulerTimingMetrics::default();
+                        let res = receive_and_buffer.receive_and_buffer_packets(
+                            &mut container,
+                            &mut timing_metrics,
+                            &mut count_metrics,
+                            &decision,
+                        );
```

Comment:
Note to self; this call takes some time, maybe can take shortcut, just dump packets into container?